src/content/guides/upgrade-tiptap-v2.mdx

```diff
@@ -138,6 +138,34 @@ new Editor({
 })
 ```
 
+### Introducing `shouldRerenderOnTransaction`
+
+The `@tiptap/react` Editor configuration now includes `shouldRerenderOnTransaction`, controlling component rerendering.
+
+Previously (pre-3.0.0), the Editor rerendered on every transaction-induced state update, facilitating continuous editor object tracking but potentially impacting rendering performance.
+
+Tiptap 3.0.0 enables `shouldRerenderOnTransaction` by default. Disable it by setting `shouldRerenderOnTransaction: false` in the editor configuration.
+
+Post-upgrade, UI elements relying on the Editor's state might become unresponsive. Solutions include disabling the option (potential performance implications) or implementing manual state tracking, as demonstrated:
+
+```js
+function MyEditor() {
+  const [selection, setSelection] = useState({ from: 0, to: 0 });
+
+  const editor = useEditor({
+    // Editor configuration
+    onTransaction({ transaction }) {
+      setSelection({
+        from: transaction.selection.from,
+        to: transaction.selection.to,
+      });
+    },
+  });
+}
+```
+
+This allows explicit state extraction and management for dependent UI logic.
```

Comment:
Isn't it better to use the `useEditorState` hook to subscribe to the selection state, instead of duplicating the state with the useState hook? https://github.com/ueberdosis/tiptap/blob/develop/packages/react/src/useEditorState.ts

Reply:
It depends I think. If you want the whole editor state then yes but if you only want to extract specific values you could do it directly on the useEditor hook itself.

We could mention it though.

Reply:
The useEditorState has a selector function that lets you only react to changes in specific values of the Editor state